src/bootstrap/bootstrap.py

```diff
@@ -587,6 +596,61 @@ def fix_bin_or_dylib(self, fname):
             print("warning: failed to call patchelf:", reason)
             return
 
+    # Return the stage1 compiler to download, if any.
+    def maybe_download_stage1(self):
+        # If `download-stage1` is not set, default to rebuilding.
+        if self.get_toml("download-stage1", section="rust") != "true":
+            return None
+        # Look for a version to compare to based on the current commit.
+        # There are a few different cases to handle.
+        # 1. This commit is a fast-forward from master: `master - * - * - HEAD`
+        # 2. This commit and master have diverged:
+        # ```
+        #   Y - * - HEAD
+        #  /
+        # X - * - master
+        # ```
+        # In this case, we should compare to `X`.
+        # 3. `master` and `HEAD` are radically different (>100 commits, or similar). This probably
+        # means that `master` does *not* correspond to the version we want to compare to, e.g. a
+        # fork. Instead, we want to compare to `rust-lang/rust:master`, which this has to share a
+        # recent merge base with.
+
+        # Find which remote corresponds to `rust-lang/rust`.
+        remotes = subprocess.check_output(["git", "remote", "-v"], universal_newlines=True)
+        # e.g. `origin https://github.com//rust-lang/rust (fetch)`
+        rust_lang_remote = next(line for line in remotes.splitlines() if "rust-lang/rust" in line)
+        rust_lang_remote = rust_lang_remote.split()[0]
+
+        # Find which commit to compare to
+        merge_base = ["git", "merge-base", "HEAD", "{}/master".format(rust_lang_remote)]
+        commit = subprocess.check_output(merge_base, universal_newlines=True).strip()
```

Comment:
merge-base is probably overcomplicated. I expect we can just pull the most recent bors-authored commit, like we do for the LLVM downloads (but without a file filter; it's not worth trying to figure out which things can alter compiler behavior and what can't).

Reply:
merge-base seems a lot more accurate than checking for the author though ... is there a reason you think it's more complicated than `git log --author bors` or something like that?

Reply:
it's more accurate but is also not going to work well on e.g. non-master branches. I'd rather just stick with bors author for now and if we run into problems then we can explore switching.

Reply:
Ok, I switched this to use `--author=bors`.